magma/passes/unused_undriven.py

```diff
@@ -0,0 +1,32 @@
+from abc import abstractmethod
+from .passes import CircuitPass
+
+
+class UndrivenUnusedPass(CircuitPass):
```

Comment:
I like moving this to `passes.py` as follows:

```python
class EditCircuitPass(CircuitPass):
    @abstractmethod
    def edit(self, circuit):
        raise NotImplementedError()

    def __call__(self, circuit):
        with circuit.open():
            self.edit(circuit)
```

and then having the following:

```python
def _drive_undriven(interface):
    undriven = False
    for port in interface.ports.values():
        if port.is_input() and port.trace() is None:
            undriven = True
            port.undriven()


def _terminate_unused(interface):
    terminated = False
    for port in interface.ports.values():
        if port.is_output() and port.wired() is False:
            terminated = True
            port.unused()


class DriveUndrivenPass(EditCircuitPass):
    def edit(self, circuit):
        if _drive_undriven(circuit.interface):
            circuit._is_definition = True
        for inst in circuit.instances:
            _drive_undriven(inst.interface)


class TerminateUnusedPass(EditCircuitPass):
    def edit(self, circuit):
        if _terminate_unused(circuit.interface):
            circuit._is_definition = True
        for inst in circuit.instances:
            _terminate_unused(inst.interface)
```

If possible I would find a better name than `unused_undriven.py`. I also think it's ok to have 2 files: `drive_undrive.py` and `terminate_unused.py`.